assets/jsx/bagger.jsx

```diff
@@ -1,70 +1,96 @@
         const dataset = b.dataset
         const args = [dataset.accessKeyId, dataset.secretAccessKey, dataset.bucket, dataset.region,
                         dataset.keyPrefix]
         if (args.some(arg => arg !== undefined)) {
-            actions.updateConfig(...args)
+            configurationUpdated(...args)
         }
-        actions.testConfiguration()
-
-        actions.createHasher()
+        testConfiguration(...args)
     }
 
-    render() {
-        const {bagger, hasher, uploader, actions} = this.props;
-
-        const bytesUploaded = [...uploader.bytesUploaded.values()].reduce((r, n) => r + n, 0);
-        const bytesHashed = [...hasher.bytesHashed.values()].reduce((r, n) => r + n, 0);
-        const totalBytes = [...bagger.sizes.values()].reduce((r, n) => r + n, 0);
-
-        const {totalHashers, activeHashers} = hasher.hasherStats
+    render(): ?React.Element {
+        // TODO: move calculations into stores
+        const bytesHashed = [...this.state.bytesHashed.values()].reduce((r, n) => r + n, 0);
+        const totalBytes = [...this.state.bagFiles.values()].reduce((r, f) => r + f.fileSize, 0);
+        const {upload, bagFiles} = this.state
+        const bytesUploaded = [...this.state.upload.bytesUploaded.values()].reduce((r, n) => r + n, 0);
 
         return (
             <div className="bagger">
-                <ServerInfo
-                    uploader={uploader}
-                    updateConfig={actions.updateConfig}
-                    testConfiguration={actions.testConfiguration}
+                <ServerInfo uploader={upload}
+                    updateConfig={configurationUpdated}  testConfiguration={testConfiguration}
                 />
-                {uploader.configStatus.message === 'OK' && (
+                {upload.message === 'OK' && (
                     <div>
-                        <SelectFiles onFilesChange={(files) => actions.addFiles(files)} />
-                        {bagger.files.size > 0 && (
-                            <div className="dashboard well well-sm clearfix">
-                                <div className="col-sm-6 hash-stats">
-                                    <h5>Hashing</h5>
-                                    <Progress current={bytesHashed} total={totalBytes} />
-                                    <Throughput current={bytesHashed} total={totalBytes} />
-                                    {activeHashers} of {totalHashers} hashers are active.
-                                </div>
-                                <div className="col-sm-6 upload-stats">
-                                    <h5>Uploads</h5>
-                                    <Progress current={bytesUploaded} total={totalBytes} />
-                                    <Throughput current={bytesUploaded} total={totalBytes} />
+                        <SelectFiles onFilesSelected={(files) => filesSelected(files)} />
+                        {bagFiles.size > 0 && (
```

Comment:
Probably out of scope for this PR but these large blocks are making me wonder whether we should refactor this into separate component(s) which could handle the state-dependent displays for progress.